dagobah/core/job.py

```diff
@@ -474,3 +477,55 @@ def destroy_snapshot(self):
         """ Destroy active copy of the snapshot """
         logger.debug('Destroying DAG snapshot for job {0}'.format(self.name))
         self.snapshot = None
+
+    def verify(self, context=None):
+        """
+        Verify that the job has no cycles where a JobTask circularly
+        references another JobTask so that we know we can safely snapshot
+        the DAG.
+        Explanation:
+            1. If the current job is in the context, the job is not valid
+            2. Perform a topological sort without expanding tasks (current job
+               is acyclic)
+            3. Traverse nodes in topological order. For each node that is a
+               Job, run verify on that node, passing in the current context.
+        """
```

Comment:
Could you clarify that this returns a bool? When I read that something is going to "verify" something I think it'll throw an exception if the invariant doesn't hold